source/common/http/http3/conn_pool.h

```diff
@@ -26,6 +37,41 @@ class ActiveClient : public MultiplexedActiveClientBase {
                                     data) {}
 };
 
+// Http3 subclass of FixedHttpConnPoolImpl which exists to store quic data.
+class Http3ConnPoolImpl : public FixedHttpConnPoolImpl {
+public:
+  Http3ConnPoolImpl(Upstream::HostConstSharedPtr host, Upstream::ResourcePriority priority,
+                    Event::Dispatcher& dispatcher,
+                    const Network::ConnectionSocket::OptionsSharedPtr& options,
+                    const Network::TransportSocketOptionsSharedPtr& transport_socket_options,
+                    Random::RandomGenerator& random_generator,
+                    Upstream::ClusterConnectivityState& state, CreateClientFn client_fn,
+                    CreateCodecFn codec_fn, std::vector<Http::Protocol> protocol,
+                    TimeSource& time_source)
+      : FixedHttpConnPoolImpl(host, priority, dispatcher, options, transport_socket_options,
+                              random_generator, state, client_fn, codec_fn, protocol) {
+    auto source_address = host_->cluster().sourceAddress();
+    if (!source_address.get()) {
+      auto host_address = host->address();
+      source_address = Network::Utility::getLocalAddress(host_address->ip()->version());
+    }
+    Network::TransportSocketFactory& transport_socket_factory = host->transportSocketFactory();
+    quic_info_ = std::make_unique<Quic::PersistentQuicInfoImpl>(
+        dispatcher, transport_socket_factory, time_source, source_address);
+    setQuicConfigFromClusterConfig(host_->cluster(), quic_info_->quic_config_);
```

Comment:
It looks like this method is only called from the constructor. Do we have other places we expect to call it from? If not, perhaps we should just inline it into the constructor?

Reply:
I think it's slightly cleaner as a utility, but I could make it a static method in the pool to make usage a bit more clear?